monai/metrics/utils.py

```diff
@@ -292,45 +323,410 @@ def prepare_spacing(
+        [[-0.25, 0.0, 0.25], [-0.25, 0.0, 0.25], [0.125, -0.125, -0.125], zeros],
+        [[0.0, -0.25, -0.25], [0.0, 0.25, 0.25], [-0.125, 0.125, 0.125], zeros],
+        [[-0.125, 0.125, 0.125], [0.125, -0.125, -0.125], zeros, zeros],
+        [[-0.125, -0.125, -0.125], [-0.25, -0.25, -0.25], [0.25, 0.25, 0.25], [0.125, 0.125, 0.125]],
+        [[-0.125, -0.125, 0.125], [0.125, -0.125, 0.125], [0.125, -0.125, -0.125], zeros],
+        [[0.0, 0.0, -0.5], [0.25, 0.25, 0.25], [-0.125, -0.125, -0.125], zeros],
+        [[0.125, -0.125, 0.125], [0.125, -0.125, -0.125], zeros, zeros],
+        [[0.0, -0.5, 0.0], [0.25, 0.25, 0.25], [0.125, 0.125, 0.125], zeros],
+        [[-0.125, -0.125, 0.125], [0.125, -0.125, -0.125], zeros, zeros],
+        [[0.0, -0.25, -0.25], [0.0, 0.25, 0.25], zeros, zeros],
+        [[0.125, -0.125, -0.125], zeros, zeros, zeros],
+        [[0.5, 0.0, 0.0], [0.5, 0.0, 0.0], zeros, zeros],
+        [[-0.5, 0.0, 0.0], [-0.25, 0.25, 0.25], [-0.125, 0.125, 0.125], zeros],
+        [[0.5, 0.0, 0.0], [0.25, -0.25, 0.25], [-0.125, 0.125, -0.125], zeros],
+        [[0.25, -0.25, 0.0], [0.25, -0.25, 0.0], zeros, zeros],
+        [[0.5, 0.0, 0.0], [-0.25, -0.25, 0.25], [-0.125, -0.125, 0.125], zeros],
+        [[-0.25, 0.0, 0.25], [-0.25, 0.0, 0.25], zeros, zeros],
+        [[0.125, 0.125, 0.125], [-0.125, 0.125, 0.125], zeros, zeros],
+        [[-0.125, 0.125, 0.125], zeros, zeros, zeros],
+        [[0.5, 0.0, -0.0], [0.25, 0.25, 0.25], [0.125, 0.125, 0.125], zeros],
+        [[0.125, -0.125, 0.125], [-0.125, -0.125, 0.125], zeros, zeros],
+        [[-0.25, -0.0, -0.25], [0.25, 0.0, 0.25], zeros, zeros],
+        [[0.125, 0.125, 0.125], zeros, zeros, zeros],
+        [[-0.25, -0.25, 0.0], [0.25, 0.25, -0.0], zeros, zeros],
+        [[0.125, 0.125, 0.125], zeros, zeros, zeros],
+        [[0.125, 0.125, 0.125], zeros, zeros, zeros],
+        [zeros, zeros, zeros, zeros],
+    ]
+    return torch.as_tensor(ret, device=device)
+
+
+def create_table_neighbour_code_to_surface_area(spacing_mm, device=None):
+    """
+    Returns an array mapping neighbourhood code to the surface elements area.
+    Adapted from https://github.com/deepmind/surface-distance
+
+    Note that the normals encode the initial surface area. This function computes
+    the area corresponding to the given `spacing`.
+
+    Args:
+        spacing_mm: a sequence of 3 numbers. Voxel spacing along the first 3 spatial axes.
+        device: device to put the table on.
+
+    Returns:
+        An array of size 256, mapping neighbourhood code to the surface area.
+        ENCODING_KERNEL[3] which is the kernel used to compute the neighbourhood code.
+    """
+    # compute the area for all 256 possible surface elements given a 2x2x2 neighbourhood according to the spacing_mm
+    c = _get_neighbour_code_to_normals_table(device)
+    s = torch.as_tensor(
+        [[[spacing_mm[1] * spacing_mm[2], spacing_mm[0] * spacing_mm[2], spacing_mm[0] * spacing_mm[1]]]],
```

Comment:
could we also has a simple check for `spacing_mm` ? I think it should has length 3.